comp/otelcol/otlp/components/exporter/logsagentexporter/logs_exporter_test.go

```diff
@@ -99,6 +99,40 @@ func TestLogsExporter(t *testing.T) {
 			},
 			expectedTags: [][]string{{"otel_source:datadog_agent"}},
 		},
+		{
+			name: "resource-attribute-source",
+			args: args{
+				ld: func() plog.Logs {
+					l := testutil.GenerateLogsOneLogRecord()
+					rl := l.ResourceLogs().At(0)
+					resourceAttrs := rl.Resource().Attributes()
+					resourceAttrs.PutStr("datadog.log.source", "custom_source_rattr")
+					return l
+				}(),
+				otelSource:    otelSource,
+				logSourceName: "custom_source_rattr",
```

Comment:
It looks like setting this field forces log source to be `custom_source_rattr`, regardless of the attribute `datadog.log.source`, so this test may not really validate if `datadog.log.source` works

https://github.com/DataDog/datadog-agent/blob/fe16b1d2d98595695e79f269fb1db68198771544/comp/otelcol/otlp/components/exporter/logsagentexporter/logs_exporter_test.go#L267

Suggest to change `logSourceName` back to `otlp_log_ingestion` and then see if `datadog.log.source` takes precedence.  Same for the other test.

Reply:
Good call, updated the test 👍